### PR TITLE
Error handling on category creation

### PR DESCRIPTION
**Pull request by felabel**
This is linked to issue - https://github.com/ushahidi/platform/issues/4751

The category creation process wasn't handling errors properly. This could be a backend error, i noticed this when I altered the API for categories to see the behaviour for the user. There was no error response so the user would know the next steps or what is happening.

I added a check for a 404 error for better user experience and accessibility.

### CODE REVIEW

**Comment by Angamanga** [comment]
Thank you for your PR @felabel  🙏. Could you please add a short description and a checklist for testing the fix as well and then we'll review it 👍

**Comment by felabel** [comment]
@Angamanga please let me know if this is enough

To replicate;

1. Go to settings/categories through the side bar
2. Create a new category and save
if the process fails (this process failing scope assumes that there might be an error from the backend, server issues or API related issues) What I did was to cause a typo error in the API request and noticed the error handling defect
This would also cover if there is an network error

Expected Result - There should be an error response so the user knows what is wrong
Actual result - on error, nothing happens on the UI, user is left confused on next steps.

File: apps/web-mzima-client/src/app/shared/components/collections/collections.component.html [hunk 1]
```diff
@@ -93,6 +93,10 @@
         </app-collection-item>
       </div>
     </div>
+
+    <mat-dialog-actions align="end">
+      <mzima-client-button (buttonClick)="closeModal()"> Done </mzima-client-button>
+    </mat-dialog-actions>
   </ng-container>
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/shared/components/collections/collections.component.html:100]: Is this change connected to this issue (https://github.com/ushahidi/platform/issues/4751)?

File: apps/web-mzima-client/src/app/shared/components/collections/collections.component.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ import {
   PostResult,
   AccountNotificationsInterface,
 } from '@mzima-client/sdk';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { AlertService } from 'libs/sdk/src/lib/services/alerts.service';
 import { BaseComponent } from '../../../base.component';
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/shared/components/collections/collections.component.ts:19]: You don't need to add a new alert-service, we already have a service called notification-service you can use, look for how that is used in the rest of the codebase: https://github.com/ushahidi/platform-client-mzima/blob/development/apps/web-mzima-client/src/app/core/services/notification.service.ts

File: apps/web-mzima-client/src/app/settings/categories/create/create.component.ts [hunk 3]
```diff
@@ -21,6 +28,11 @@ export class CreateComponent {
         this.router.navigate(['settings/categories']);
       },
       error: ({ error }) => {
+        if (error.errors[0].status === 404) {
+          this.notificationService.showError(
+            this.translate.instant('An error occured, please try again'),
+          );
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/settings/categories/create/create.component.ts:34]: Is there a way to show the error-message as well, for example, look at how its handled here: https://github.com/ushahidi/platform-client-mzima/blob/development/apps/web-mzima-client/src/app/post/post-edit/post-edit.component.ts#L644

**Review by Angamanga — CHANGES_REQUESTED** [review]
Good start @felabel! See my comments in the code for changes :)

File: apps/web-mzima-client/src/app/shared/components/collections/collections.component.html (hunk 1 above)
Comment (felabel) [reply apps/web-mzima-client/src/app/shared/components/collections/collections.component.html:100]: no this isn't related, i forgot to checkout to a new branch. this change is related to the task on adding posts to a collection. I made a PR for it. @Angamanga

**Comment by MugoBrian** [comment]
Hello @felabel , is the fix for this issue still pending? Since I am seeing there's an unhandled change requested.

**Pull request closed without merging**